Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-119.ca-central-1.compute.internal
Sha: b46c239767a843cbd756a02970e4a390c456243e
Number of Error(s): 8
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-13845/HPCC-Platform/esp/services/ws_dfu/ws_dfuService.cpp:1632:30: error: \xe2\x80\x98CDFUChangeRestriction_Restricted\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-13845/HPCC-Platform/esp/services/ws_dfu/ws_dfuService.cpp:2202:34: error: \xe2\x80\x98CDFUChangeRestriction_Restricted\xe2\x80\x99 was not declared in this scope
make[2]: *** [esp/services/ws_dfu/CMakeFiles/ws_dfu.dir/build.make:154: esp/services/ws_dfu/CMakeFiles/ws_dfu.dir/ws_dfuService.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:10614: esp/services/ws_dfu/CMakeFiles/ws_dfu.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

[Build error:](https://gist.github.com/9a2fbb3344f4a8afbf97188bed575891)
Build log [Build log](https://gist.github.com/HPCCSmoketest/9a2fbb3344f4a8afbf97188bed575891#file-result-20-06-10-11-36-16-log)

--- a/esp/services/ws_dfu/ws_dfuService.cpp
+++ b/esp/services/ws_dfu/ws_dfuService.cpp
@@ -1629,7 +1629,8 @@ bool CWsDfuEx::changeFileRestrictions(IEspContext &context, IEspDFUArrayActionRe
     if (changeRestriction == CDFUChangeRestriction_NoChange)
         return true;
 
-    context.ensureFeatureAccess("DFURestrictedAccess", SecAccess_Write, ECLWATCH_DFU_ACCESS_DENIED, "DFURestrictedAccess: Permission denied.");
+    if (changeRestriction == CDFUChangeRestriction_NotRestricted)
+        context.ensureFeatureAccess("CodeSignAccess", SecAccess_Full, ECLWATCH_DFU_ACCESS_DENIED, "CodeSignAccess: Permission denied.");
 
     if (isDetachedFromDali())
         throw makeStringException(ECLWATCH_INVALID_INPUT, "ESP server is detached from Dali. Please try later.");
@@ -2198,7 +2199,8 @@ void CWsDfuEx::doGetFileDetails(IEspContext &context, IUserDescriptor *udesc, co
     }
     if (changeRestriction != CDFUChangeRestriction_NoChange)
     {
-        context.ensureFeatureAccess("DFURestrictedAccess", SecAccess_Write, ECLWATCH_DFU_ACCESS_DENIED, "DFURestrictedAccess: Permission denied.");
+        if (changeRestriction == CDFUChangeRestriction_NotRestricted)
+            context.ensureFeatureAccess("CodeSignAccess", SecAccess_Full, ECLWATCH_DFU_ACCESS_DENIED, "CodeSignAccess: Permission denied.");
         df->setRestrictedAccess(changeRestriction==CDFUChangeRestriction_Restrict);
     }
 

